lib/_stream_readable.js

```diff
@@ -660,7 +661,7 @@ Readable.prototype.on = function(ev, fn) {
     this.resume();
   }
 
-  if (ev === 'readable' && this.readable) {
+  if (ev === 'readable') {
```

Comment:
If that can be a problem, then we need to add tests for that or tweak that relevant code possibly.